cmd/spire-server/cli/validate/validate.go

```diff
@@ -0,0 +1,74 @@
+package validate
+
+import (
+	"flag"
+
+	"github.com/mitchellh/cli"
+	"github.com/spiffe/spire/cmd/spire-server/cli/run"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+)
+
+const defaultConfigPath = "server.conf"
+
+func NewValidateCommand() cli.Command {
+	return newValidateCommand(common_cli.DefaultEnv)
+}
+
+func newValidateCommand(env *common_cli.Env) *validateCommand {
+	return &validateCommand{
+		env: env,
+	}
+}
+
+type validateCommand struct {
+	env *common_cli.Env
+
+	configPath string
+	ExpandEnv  bool
+}
+
+func (c *validateCommand) Help() string {
+	// ignoring parsing errors since "-h" is always supported by the flags package
+	_ = c.parseFlags([]string{"-h"})
+	return ""
+}
+
+func (c *validateCommand) Synopsis() string {
+	return "Validates a SPIRE server configuration file"
+}
+
+func (c *validateCommand) Run(args []string) int {
+	if err := c.parseFlags(args); err != nil {
+		return 1
+	}
+	if err := c.run(); err != nil {
+		// Ignore error since a failure to write to stderr cannot very well be reported
+		_ = c.env.ErrPrintf("SPIRE server configuration file is invalid: %v\n", err)
+		return 1
+	}
+	if err := c.env.Println("SPIRE server configuration file is valid."); err != nil {
+		return 1
```

Comment:
I don't think we should exit 1 if we know the configuration file is valid. Printing "valid" can be best effort.

```go
_ = c.env.Println("SPIRE server configuration file is valid.")
return 0
```

Reply:
I fixed it with https://github.com/spiffe/spire/pull/1471/commits/5a58b27950b3f69c7baca6c60c78c305caa1f3fe.